### PR TITLE
Accept adapter/debugee with different names in launch.json

### PR DESCRIPTION
**Pull request by weihanglo**
In a debugee configurations, the `type` field refers to an adapter
defined in `dap.adapters.<adapter-name>`. Current implementation has
an assumption that the `config.type`, which is the name of an adapter,
is the same as a debugee name. Therefore it is used for retrieving a
debugee configurations as a key of `dap.configurations[config.type]`.

The assumption works in most cases, since most adapter-debugee gets the
same name for which a simple "filetype" is often chosen, such as php or
python. One example breaks the rule is lldb adapter. There is a mismatch
between the adapter lldb and its multiple debugees like c++ and rust.

This commit simply groups filetypes by adapaters to fix it.

### CODE REVIEW

**Comment by weihanglo** [comment]
Also, thank you for creating such a great plugin 🎉 🎉

File: lua/dap/ext/vscode.lua [hunk 1]
```diff
@@ -16,15 +16,29 @@ function M.load_launchjs(path)
...
+
   assert(data.configurations, "launch.json must have a 'configurations' key")
   for _, config in ipairs(data.configurations) do
     assert(config.type, "Configuration in launch.json must have a 'type' key")
-    local configurations = dap.configurations[config.type]
-    if not configurations then
-      configurations = {}
-      dap.configurations[config.type] = configurations
+    local debugees = debugees_of_adapter[config.type]
+    if debugees then
```
Comment (mfussenegger) [inline lua/dap/ext/vscode.lua:35]: Doesn't this mean that if `dap.configurations` is initially empty,  nothing would be added? 

Did you see the conversations in https://github.com/mfussenegger/nvim-dap/pull/156 ?



The debugger naming is also a bit confusing to me - the introduction in dap.txt refers to the application being debugged as debugee but in this PR it's used for the configuration indices

**Review by mfussenegger — COMMENTED** [review]
Thank you for the PR, I  left some feedback

File: lua/dap/ext/vscode.lua (hunk 1 above)
Comment (weihanglo) [reply lua/dap/ext/vscode.lua:35]: Oh, sorry. I didn't notice that PR. Also, It should be filetype not debugee itself, that's my fault.  

For the improvement, we can fallback to the old way (using adapter's name as filetype to add new debugee config) if there is no debugee config for that filetype. Doing it this way we need not to add any new mapping key between launch.json and nvim-dap.  What do you think?

Comment (mfussenegger) [reply lua/dap/ext/vscode.lua:35]: > For the improvement, we can fallback to the old way (using adapter's name as filetype to add new debugee config) if there is no debugee config for that filetype. Doing it this way we need not to add any new mapping key between launch.json and nvim-dap. What do you think?

Sounds good

Comment (weihanglo) [reply lua/dap/ext/vscode.lua]: Just fixed it. Let me know if anything need to improve 🙂

**Comment by mfussenegger** [comment]
Thanks :+1: 

I merged the change but squashed the commits and reworded the commit message a bit.

**Pull request merged**